signer-service/src/api/services/transactions/onrampTransactions.ts

```diff
@@ -20,186 +36,136 @@ export async function prepareOnrampTransactions(
+    const accountNetworkId = getNetworkId(account.network);
+
+    if (accountNetworkId === getNetworkId(Networks.Moonbeam)) {
+      const moonbeamEphemeralStartingNonce = 0;
+      const moonbeamToPendulumXCMTransaction = await createMoonbeamToPendulumXCM(
+        pendulumEphemeralEntry.address,
+        inputAmountRaw,
+        inputTokenDetails.moonbeamErc20Address,
+      );
+      unsignedTxs.push({
+        tx_data: encodeSubmittableExtrinsic(moonbeamToPendulumXCMTransaction),
+        phase: 'moonbeamToPendulumXCM',
+        network: account.network,
+        nonce: moonbeamEphemeralStartingNonce,
+        signer: account.address,
       });
 
+      if (toNetworkId !== getNetworkId(Networks.AssetHub)) {
+        if (!isEvmTokenDetails(outputTokenDetails)) {
+          throw new Error(`Output token must be an EVM token for onramp to any EVM chain, got ${quote.outputCurrency}`);
+        }
+        const { approveData, swapData } = await createOnrampSquidrouterTransactions({
+          outputTokenDetails,
+          toNetwork,
+          rawAmount: outputAmountRaw,
+          addressDestination: account.address,
+          fromAddress: account.address,
+          moonbeamEphemeralStartingNonce: moonbeamEphemeralStartingNonce + 1,
+        });
+
+        unsignedTxs.push({
+          tx_data: encodeEvmTransactionData(approveData.data),
+          phase: 'moonbeamSquidrouter',
+          network: account.network,
+          nonce: moonbeamEphemeralStartingNonce + 1,
+          signer: account.address,
+        });
+        unsignedTxs.push({
+          tx_data: encodeEvmTransactionData(swapData.data),
+          phase: 'moonbeamSquidrouter',
+          network: account.network,
+          nonce: moonbeamEphemeralStartingNonce + 2,
+          signer: account.address,
+        });
+      }
+    } else if (accountNetworkId === getNetworkId(Networks.Pendulum)) {
+      const { approveTransaction, swapTransaction } = await createNablaTransactionsForQuote(quote, account);
+
       unsignedTxs.push({
-        tx_data: encodeEvmTransactionData(approveData),
-        phase: 'moonbeam', // TODO assign correct phase
+        tx_data: approveTransaction,
+        phase: 'nablaApprove',
         network: account.network,
         nonce: 0,
         signer: account.address,
       });
+
       unsignedTxs.push({
-        tx_data: encodeEvmTransactionData(swapData),
-        phase: 'moonbeam', // TODO assign correct phase
+        tx_data: swapTransaction,
+        phase: 'nablaSwap',
         network: account.network,
         nonce: 0,
         signer: account.address,
       });
-    } else if (accountNetworkId === getNetworkId(Networks.Pendulum)) {
-      // TODO implement creation of unsigned ephemeral txs for swaps
 
-      if (toNetworkId === getNetworkId(Networks.AssetHub)) {
-        // TODO implement creation of unsigned ephemeral tx for Pendulum -> AssetHub
+      if (toNetworkId !== getNetworkId(Networks.AssetHub)) {
```

Comment:
I think I'd flip this condition and make it 
```suggestion
      if (toNetworkId === getNetworkId(Networks.AssetHub)) {
      // Do the Assethub stuff
      } else { // Do Moonbeam stuff }
```